packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -1063,4 +1097,146 @@ export default class MuiDetailGenerator implements DetailGenerator {
+
+      if (ast) {
+        const transformIdAttribute = (node: ts.Node) => {
+          const attributeName = 'id';
+          if (isJsxAttributeWithName(node, attributeName)) {
+            return createStringJsxAttribute(attributeName, propName);
+          }
+        };
+  
+        const transformLabelAttribute = (node: ts.Node) => {
+          const attributeName = 'label';
+          if (isJsxAttributeWithName(node, attributeName)) {
+            return createStringJsxAttribute(attributeName, propName);
+          }
+        };
+  
+        const transformValueAttribute = (node: ts.Node) => {
+          const attributeName = 'value';
+          if (isJsxAttributeWithName(node, attributeName)) {
+            return this.createValueJsxAttribute(propName, propType);
+          }
+        };
+
+        const transformOnChangeAttribute = (node: ts.Node) => {
+          const attributeName = 'onChange';
+          if (isJsxAttributeWithName(node, attributeName)) {
+            return this.createFormikOnChangeAttribute(propName);
+          }
+        };
+
+        const transformInputTemplate = (node: ts.Node) => {
+          return transformIdAttribute(node) || transformLabelAttribute(node) || transformValueAttribute(node) || transformOnChangeAttribute(node);
+        };
+
+        const transformationResult = ts.transform(ast, [transformer(transformInputTemplate)]);
+        return printSourceCode(transformationResult.transformed[0]);
+      }
+    }
+  };
+
+  private createValueJsxAttribute(
+    name: string,
+    propType: PropertyType
+  ): ts.JsxAttribute {
+    // if (this._context.formatter === Formatter.ReactIntl) {
+    //   if (type === InputType.date) {
+    //     return this.createDateValueFormattedAttribute(name);
+    //   } else {
+    //     return this.createTextValueFormattedAttribute(name);
+    //   }
+    // } else {
+    //   return this.createTextValueAttribute(name);
+    // }
+
+    return this.createFormikValueAttribute(name);
+  }
+
+  private createFormikValueAttribute(name: string): ts.JsxAttribute {
+    return factory.createJsxAttribute(
+      factory.createIdentifier("value"),
+      factory.createJsxExpression(
+        undefined,
+        factory.createPropertyAccessExpression(
+          factory.createPropertyAccessExpression(
+            factory.createIdentifier("formik"),
+            factory.createIdentifier("values")
+          ),
+          factory.createIdentifier(name)
+        )
+      )
+    );
+  }
+
+  private createFormikOnChangeAttribute(name: string): ts.JsxAttribute {
+    return factory.createJsxAttribute(
+      factory.createIdentifier("onChange"),
+      factory.createJsxExpression(
+        undefined,
+        factory.createPropertyAccessExpression(
+          factory.createIdentifier("formik"),
+          factory.createIdentifier("handleChange")
+        )
+      )
+    );
+  }
+
+  private createGridItemElement(children: readonly ts.JsxChild[]): ts.JsxElement {
```

Comment:
we try to avoid `factory.create()` (that is the idea of the templates)

there are multiple cross cutting concerns:
* input
* grid
* text messages
* form binding (formik)